src/core/thread.d

```diff
@@ -4427,13 +4430,31 @@ private:
             {
                 m_ctxt.bstack = m_pmem + sz;
                 m_ctxt.tstack = m_pmem + sz;
+                void* guard = m_pmem;
             }
             else
             {
                 m_ctxt.bstack = m_pmem;
                 m_ctxt.tstack = m_pmem;
+                void* guard = m_pmem + sz - PAGESIZE;
             }
             m_size = sz;
+
+            static if( __traits( compiles, mmap ) )
+            {
+                // Mark end of stack
+                for ( ubyte* g = cast(ubyte*)guard; g < guard + PAGESIZE; g+= 32)
+                    g[0 .. 32] = cast(ubyte[]) "END OF FIBER -- END OF FIBER -- ";
```

Comment:
Not going to rediscuss why Exceptions for async failures are wrong, unwinding is hardly possible for those, and having the compiler assume that each instruction could throw requires complex unwind information or conservative (slow) code.
In any case, neither an exception nor a segfault for a stack overflow tells you much else that that the stack overflowed.